server/models/dataset.py

```diff
@@ -327,16 +296,49 @@ def reviewImages(self, dataset, acceptedImages, flaggedImages, user):
                 Folder.countFolders(prereviewFolder)) == 0:
             Folder.remove(prereviewFolder)
 
-    def registerMetadata(self, dataset, csvFile):
+    def registerMetadata(self, dataset, csvFile, user, sendMail=False):
         """Register a .csv file containing metadata about images."""
-        # Add image metadata information to list
-        registeredMetadata = dataset['meta'].get('metadata', [])
-        registeredMetadata.append({
+        # Check if image metadata is already registered
+        if self.findOne({'meta.metadataFiles.fileId': csvFile['_id']}):
+            raise ValidationException('Metadata file is already registered.')
+
+        # Add image metadata file information to list
+        now = datetime.datetime.utcnow()
+        metadataFiles = dataset['meta'].get('metadataFiles', [])
+        metadataFiles.append({
             'fileId': csvFile['_id'],
-            'time': datetime.datetime.utcnow()
+            'user': user['_id'],
```

Comment:
Yeah, `user` is an important addition. Let's make it `userId`, just for consistency with stuff like `creatorId`.